on-chain/EvmRouter.sol

```diff
@@ -0,0 +1,54 @@
+//SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.15;
+
+interface IMemoInteractor {
+  event MemoInteraction(bytes16 indexed memo);
+}
+
+interface IRouting is IMemoInteractor {
+  error WormholeInteractionFailed(bytes lowLevelData);
+  error TokenNotRegistered(bytes20 addressOrTokenNumber);
+  error SenderIsNotOwner(address sender, address owner);
+
+  function swimUsdAddress() external view returns (address);
+
+  function onChainSwap(
+    address fromToken,
+    uint inputAmount,
+    address toOwner,
+    address toToken,
+    uint minimumOutputAmount,
+    bytes16 memo
+  ) external returns (uint outputAmount);
+
+  function propellerInitiate(
+    address fromToken,
+    uint inputAmount,
+    uint16 wormholeRecipientChain,
+    bytes32 toOwner,
+    bool gasKickstart,
+    uint64 maxPropellerFee,
+    uint16 toTokenNumber,
+    bytes16 memo
+  ) external payable returns (uint swimUsdAmount, uint64 wormholeSequence);
+
+  function propellerInitiate(
+    address fromToken,
+    uint inputAmount,
+    uint16 wormholeRecipientChain,
+    bytes32 toOwner,
+    bool gasKickstart,
+    uint64 maxPropellerFee,
+    uint16 toTokenNumber,
+    uint32 wormholeNonce,
```

Comment:
When do we use this?

Reply:
This is relevant for upcoming Batch VAAs:
All messages emitted by a single transaction will go into a batch, but also all messages with the same nonce will get batched too. So this essentially allows you to group your emitted messages together in case you want to create multiple batches from a single transaction - see:
https://book.wormhole.com/technical/evm/coreLayer.html